src/backbone.virtual-collection.js

```diff
@@ -71,23 +72,32 @@ var VirtualCollection = Backbone.VirtualCollection = Backbone.Collection.extend(
       this.listenTo(this.collection, eventName, _.partial(this.trigger, eventName));
     }, this));
   },
+  
+  _clearCache: function(){
+    this._changeCache = {
+      added: [],
+      removed: [],
+      merged: []
+    };
+  },
 
   _onUpdate: function (collection, options) {
-    this.trigger('update', this, options);
+    var newOptions = _.extend({}, options, {changes: this._changeCache});    
```

Comment:
I noticed you're only passing the changes cache on the `update` event. is that how backbone behaves?